### PR TITLE
C# 10 Program.cs

### PR DESCRIPTION
**Pull request by codyconfer**
- Abstract bot logic
- use host builder
- add csharp 10 features to program.cs
- clean up using statements

### CODE REVIEW

File: src/Quotebot/Bot.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                var token = await authContext.AcquireTokenAsync(resource, credential);
+
+                return token.AccessToken;
+            });
+            // Get the API key out of the vault
+            return (await keyVault.GetSecretAsync(keyUrl)).Value;
+        }
+
+        public static async Task LogMessage(LogMessage log) =>
+            await Task.Run(() => Console.WriteLine(log.Message));
```
Comment (CraigHead) [inline src/Quotebot/Bot.cs:88]: Why run this on another thread?

File: src/Quotebot/Services/CommandsHandlerService.cs [hunk 2]
```diff
@@ -1,12 +1,10 @@
...
-using Discord.Commands;
-using Discord.WebSocket;
+﻿using Discord.Commands;
 using Microsoft.Extensions.DependencyInjection;
 using System.Reflection;
 
 namespace Quotebot.Services
 {
-    internal class CommandsHandlerService
+    public class CommandsHandlerService
```
Comment (CraigHead) [inline src/Quotebot/Services/CommandsHandlerService.cs:7]: why did this have to be public?

**Review by CraigHead — APPROVED** [review]
I approve but curious about a few things.

File: src/Quotebot/Bot.cs (hunk 1 above)
Comment (CraigHead) [inline src/Quotebot/Bot.cs:88]:
```suggestion
        public static async Task LogMessage(LogMessage log) 
        { 
            Console.WriteLine(log.Message);
            await Task.CompletedTask;
         }
```

Comment (codyconfer) [reply src/Quotebot/Bot.cs:88]: No reason in particular, was just in the mindset that we may use a more intensive logging solution at some point.

File: src/Quotebot/Services/CommandsHandlerService.cs (hunk 2 above)
Comment (codyconfer) [reply src/Quotebot/Services/CommandsHandlerService.cs:7]: DI gets upset if a public class tries to inject an internal class, so we can go with internal, but we'd have to make sure every class that consumes this is internal

Comment (codyconfer) [reply src/Quotebot/Services/CommandsHandlerService.cs:7]: Specifically, Bot.cs was upset that CommandsHanderService is internal because Bot is public

**Review by CraigHead — APPROVED** [review]

**Pull request merged**